jung-visualization/src/main/java/edu/uci/ics/jung/visualization/transform/Lens.java

```diff
@@ -34,25 +32,16 @@
 
   protected float magnification = 0.7f;
 
-  /**
-   * Create an instance with a possibly shared transform.
-   *
-   * @param component the component used for rendering
-   */
-  public Lens(Component component) {
-    setComponent(component);
-    component.addComponentListener(new ComponentListenerImpl());
+  /** @param d the size used for the lens */
+  public Lens(Dimension d) {
+    setSize(d);
   }
 
-  /**
-   * Set values from the passed component.
-   *
-   * @param component the component used for rendering
-   */
-  private void setComponent(Component component) {
-    Dimension d = component.getSize();
+  /** @param d the size used for the lens */
+  public void setSize(Dimension d) {
+
     if (d.width <= 0 || d.height <= 0) {
-      d = component.getPreferredSize();
+      d = new Dimension(600, 600);
```

Comment:
Why are we hardwiring this size?  Wouldn't it make more sense to just throw IllegalArgumentException (via Precondition) if the dimension passed in is invalid?

Reply:
that would be better to give the user a stack trace instead of just making the best of it. :)

Reply:
Actually, I do think that would be better.  I'm not a big fan of magically assuming defaults unless there's a principled way of doing so.  If the user doesn't specify what size they want their visualization to be, seems like it's not our job to pick a size for them, but to let them know that they need to.

Reply:
you are a glass half empty kind of guy! :)

Reply:
we do declare a single DEFAULT_SIZE now for the view, and that was always the same as the default for the layout algorithm (and all the other places that held that same value). I updated the code to use the VisualizationServer.DEFAULT_SIZE to tighten it all up.

Reply:
Oh, I'm worse than a glass half-empty person: I don't want us to put anything in _anyone's_ glass unless they say how much they want.  ;)

One reason that I don't want us providing a default is that I don't want people to get broken by us deciding that we want to change the default, which we might reasonably do: 600x600 is a much smaller piece of someone's screen real estate than it was 14 years ago.

I'm not going to block this PR on making this change, but I've filed #161 to sort it out.